app/api/api_v1/routers/unfccc_ingest.py

```diff
@@ -0,0 +1,391 @@
+import logging
+from typing import Union
+from sqlalchemy.orm import Session
+from app.core.aws import S3Client
+
+from fastapi import (
+    APIRouter,
+    BackgroundTasks,
+    Depends,
+    HTTPException,
+    Request,
+    UploadFile,
+    status,
+)
+from app.core.aws import S3Document
+
+from app.api.api_v1.schemas.document import (
+    BulkIngestResult,
+)
+from app.core.auth import get_superuser_details
+from app.core.aws import get_s3_client
+from app.core.ingestion.ingest_row import DocumentIngestRow, EventIngestRow
+from app.core.ingestion.pipeline import generate_pipeline_ingest_input
+from app.core.ingestion.processor import (
+    initialise_context,
+    get_dfc_ingestor,
+    get_dfc_validator,
+    get_event_ingestor,
+)
+from app.core.ingestion.reader import get_file_contents, read
+from app.core.ingestion.utils import (
+    IngestContext,
+    Result,
+    ResultType,
+    ValidationResult,
+    get_result_counts,
+)
+from app.core.ingestion.validator import validate_event_row
+from app.core.validation.types import ImportSchemaMismatchError
+from app.core.validation.util import (
+    get_new_s3_prefix,
+    write_csv_to_s3,
+    write_documents_to_s3,
+    write_ingest_results_to_s3,
+)
+from app.db.session import get_db
+
+_LOGGER = logging.getLogger(__name__)
+
+unfccc_ingest_router = r = APIRouter()
+
+
+def _start_ingest(
+    db: Session,
+    s3_client: S3Client,
+    s3_prefix: str,
+    documents_file_contents: str,
+    events_file_contents: str,
+):
+    context = None
+    # TODO: add a way for a user to monitor progress of the ingest
+    try:
+        context = initialise_context(db, "UNFCCC")
+        document_ingestor = get_dfc_ingestor(db)
+        read(documents_file_contents, context, DocumentIngestRow, document_ingestor)
+        event_ingestor = get_event_ingestor(db)
+        read(events_file_contents, context, EventIngestRow, event_ingestor)
```

Comment:
It feels like we're going to have to separate the ingestion into CCLW/UNFCCC as well - the rows are different & the handling is different too. We'll need a `collection_ingestor` for UNFCCC, the types will need to be source specific...